server/src/tests/integration/erc20.rs

```diff
@@ -0,0 +1,164 @@
+/// created tokens.
+pub async fn deploy_l1_token(from_wallet: &PrivateKeySigner, rpc_url: &str) -> Result<Address> {
+    let from_address = from_wallet.address();
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .wallet(EthereumWallet::from(from_wallet.to_owned()))
+        .on_http(Url::parse(rpc_url)?);
+
+    let contract = Erc20::deploy(
+        provider,
+        NAME.into(),
+        SYMBOL.into(),
+        from_address,
+        U256::MAX,
+    )
+    .await?;
+
+    let bridge_address = Address::from_str(&get_deployed_address("L1StandardBridgeProxy")?)?;
+    contract
+        .approve(bridge_address, U256::MAX)
+        .send()
+        .await?
+        .watch()
+        .await?;
+
+    Ok(*contract.address())
+}
+
+/// Use the `OptimismMintableERC20Factory` to create a new ERC-20 token on the L2.
+/// This token is used for bridging the L1 token with the given address to the L2.
+pub async fn deploy_l2_token(
+    from_wallet: &PrivateKeySigner,
+    l1_address: Address,
+    rpc_url: &str,
+) -> Result<Address> {
+    let factory_address = alloy::primitives::address!("4200000000000000000000000000000000000012");
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .wallet(EthereumWallet::from(from_wallet.to_owned()))
+        .on_http(Url::parse(rpc_url)?);
+
+    let contract = erc20_factory::OptimismMintableERC20Factory::new(factory_address, provider);
+    let receipt = contract
+        .createOptimismMintableERC20(l1_address, NAME.into(), SYMBOL.into())
+        .send()
+        .await
+        .unwrap()
+        .get_receipt()
+        .await
+        .unwrap();
+    let event = receipt
+        .inner
+        .logs()
+        .iter()
+        .find(|log| log.address() == factory_address)
+        .expect("OptimismMintableERC20Factory emits log");
+    let event = event
+        .log_decode::<erc20_factory::OptimismMintableERC20Factory::StandardL2TokenCreated>()
+        .expect("Event is type StandardL2TokenCreated");
+    let l2_token_address = event.data().localToken;
+    Ok(l2_token_address)
+}
+
+/// Submits a transaction to the `L1StandardBridgeProxy` to deposit L1 ERC-20 tokens (`l1_address`)
+/// into the L2, where the address of the token on the L2 is also specified (`l2_address`).
+pub async fn deposit_l1_token(
+    from_wallet: &PrivateKeySigner,
+    l1_address: Address,
+    l2_address: Address,
+    amount: U256,
+    rpc_url: &str,
+) -> Result<()> {
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .wallet(EthereumWallet::from(from_wallet.to_owned()))
+        .on_http(Url::parse(rpc_url)?);
+
+    let bridge_address = Address::from_str(&get_deployed_address("L1StandardBridgeProxy")?)?;
+    let bridge_contract = bridge::L1StandardBridge::new(bridge_address, provider);
+    let receipt = bridge_contract
+        .depositERC20(l1_address, l2_address, amount, 21_000, Default::default())
```

Comment:
I'm a bit uncomfortable about hardcoding gas limits. Are they guaranteed to be deterministic? Or maybe better to have some surplus for leeway, or even issue an estimate gas call beforehand.

Reply:
I think the surplus is acceptable here. SDKs call estimate gas before making calls, so that flow is being tested by the SDKs.